src/main/java/net/minecraftforge/event/entity/living/VillagerTradePopulationEvent.java

```diff
@@ -0,0 +1,64 @@
+package net.minecraftforge.event.entity.living;
+
+import java.util.List;
+import net.minecraft.entity.passive.EntityVillager;
+import net.minecraft.entity.passive.EntityVillager.ITradeList;
+import net.minecraftforge.fml.common.eventhandler.Cancelable;
+import net.minecraftforge.fml.common.eventhandler.Event;
+import net.minecraftforge.fml.common.eventhandler.Event.HasResult;
+import net.minecraftforge.fml.common.registry.VillagerRegistry.VillagerCareer;
+
+/**
+ * The VillagerTradePopulationEvent is fired when the trades are populated<br>
+ * in {@link EntityVillager#populateBuyingList()}.<br>
+ * It allows the modder to modify the trades of an individual Villager.<br>
+ * This event is fired via the {@link ForgeHooks#onVillagerTradePopulation}.<br>
```

Comment:
I think I mentioned it with:
> The VillagerTradePopulationEvent is fired when the trades are populated in {@link EntityVillager#populateBuyingList()}
This says, that it is called from EntityVillager.populateBuyingList()